views/index.liquid

```diff
@@ -1,37 +1,145 @@
+            {% if forloop.index == 1 %}
+              🥇
+            {% elsif forloop.index == 2 %}
+              🥈
+            {% elsif forloop.index == 3 %}
+              🥉
+            {% elsif forloop.index >= 4 %}
+              {{ forloop.index }}º
+            {% endif %}
           </th>
-          <td class="px-6 py-4">
+          <td class="w-full min-w-48 px-3 py-2 md:px-6 md:py-4">
             {{ player.player }}
           </td>
-          <td class="px-6 py-4">
-            {{ player.sum }}
+          <td class="hidden md:table-cell min-w-48 px-8 py-4">
+            {{ player.cashout }}
+          </td>
+          <td class="hidden md:table-cell min-w-48 px-8 py-4">
+            {{ player.buyin | times: -1 }}
+          </td>
+          <td class="min-w-48 px-8 py-4">
+            {{ player.nett }}
+          </td>
+          <td class="hidden md:table-cell min-w-56 px-3 py-2 md:px-6 md:py-4 text-right font-bold">
+            {{ player.percent | round: 2 }}%
+            {% if player.percent >= 90 %}
+              <svg class="h-6 w-6 inline fill-indigo-700" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+                <path d="M7,15L12,10L17,15H7Z" />
+              </svg>
+            {% elsif player.percent >= 50 and player.percent < 90 %}
+              <svg class="h-6 w-6 inline fill-blue-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+                <path d="M7,15L12,10L17,15H7Z" />
+              </svg>
+            {% elsif player.percent >= 10 and player.percent < 50 %}
+              <svg class="h-6 w-6 inline fill-green-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+                <path d="M7,15L12,10L17,15H7Z" />
+              </svg>
+            {% elsif player.percent > -10 and player.percent < 10 %}
+              <svg class="h-6 w-6 inline fill-yellow-400" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+                <path d="M19,10H5V8H19V10M19,16H5V14H19V16Z" />
+              </svg>
+            {% elsif player.percent > -50 and player.percent <= -10 %}
+              <svg class="h-6 w-6 inline fill-orange-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+                <path d="M7,10L12,15L17,10H7Z" />
+              </svg>
+            {% elsif player.percent > -90 and player.percent <= -50 %}
+              <svg class="h-6 w-6 inline fill-red-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+                <path d="M7,10L12,15L17,10H7Z" />
+              </svg>
+            {% elsif player.percent <= -90 %}
+              <svg class="h-6 w-6 inline fill-rose-700" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+                <path d="M7,10L12,15L17,10H7Z" />
+              </svg>
+            {% endif %}
           </td>
         </tr>
       {% endfor %}
     </tbody>
   </table>
 </div>
+
+<script type="text/javascript">
```

Comment:
Acredito que esse controle de qual o filtro que está selecionado poderia ser feito do lado do servidor pois fica consistente a forma de como o estado é representado. Dessa forma que está sendo executado no client (via JavaScript) o estado é "representado" depois que a página é renderizada. 

Como a query que contém o filtro é recebida pelo controller:

```TypeScript
  @Get()
  @Render('index')
  async getPlayerRanking(@Query('sortBy') sortByKey: string = 'nett') {
    return { 
      ranking: await this.appService.getPlayerRanking(sortByKey), 
      sortByKey // <---- Query passada novamente para o template
    };
  }
```

Com a query enviada novamente para o template você poderia verifcar diretamente no template no momento da renderização no servidor:

```liquid
{% if sortByKey == 'buy-in' %}
class="text-white dark:bg-gray-600 bg-blue-500 dark:hover:bg-gray-500 hover:bg-blue-100"
{% else %}
class="flex items-center justify-around w-full px-2 py-1 rounded-lg underline cursor-pointer dark:hover:text-white hover:text-blue-500"
{% endif %}
```

Ou alternativamente, dentro do próprio class alterando somente as classes que não são compartilhadas.

Um exemplo:

```liquid
class="classes-comuns {% if sortByKey == 'buy-in' %} text-white ... {% else %} text-gray-600 ... {% endif %}"
```

Dessa forma o HTML renderizado pelo servidor contém efetivamente o estado da aplicação.

* O mesmo contexto vale para definição do href (com ou sem a query)

### Referência
[HATOEAS](https://intercoolerjs.org/2016/05/08/hatoeas-is-for-humans.html)
> all “state” (that is, both the data and the network actions available on that data) is encoded in hypermedia (e.g. HTML) returned by the server.

Reply:
Feito, paizão